fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/api/villager/v1/VillagerProfessionBuilder.java

```diff
@@ -0,0 +1,106 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.villager.v1;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.item.Item;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+import net.minecraft.village.TradeOffers;
+import net.minecraft.village.VillagerProfession;
+import net.minecraft.world.poi.PointOfInterestType;
+
+import net.fabricmc.fabric.impl.object.builder.VillagerProfessionBuilderImpl;
+
+/**
+ * Allows the creation of new {@link VillagerProfession}s.
+ *
+ * <p>Note this does not register any trades to these villagers. To register trades, add a new entry with your profession as the key to {@link TradeOffers#PROFESSION_TO_LEVELED_TRADE}.
+ */
+public interface VillagerProfessionBuilder {
+	/**
+	 * Creates a builder to make a {@link VillagerProfession}.
+	 * @return A new builder.
+	 */
+	static VillagerProfessionBuilder create() {
+		return new VillagerProfessionBuilderImpl();
+	}
+
+	VillagerProfessionBuilder id(Identifier id);
+
+	/**
+	 * The {@link PointOfInterestType} the villager will search for when finding it's workstation.
+	 * @param type The {@link PointOfInterestType}.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder workstation(PointOfInterestType type);
+
+	/**
+	 * Items that a Villager may harvest in this profession.
+	 *
+	 * <p>In Vanilla this is used by the farmer to define what type of crops the farmer can harvest.
+	 *
+	 * @param items The items.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder harvestableItems(Item... items);
+
+	/**
+	 * Items that a Villager may harvest in this profession.
+	 *
+	 * <p>In Vanilla this is used by the farmer to define what type of crops the farmer can harvest.
+	 *
+	 * @param items A set of items.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder harvestableItems(Iterable<Item> items);
+
+	/**
+	 * A blocks which might suffice as a job site for a Villager.
+	 *
+	 * <p>In vanilla this is used by the {@link VillagerProfession#FARMER Farmer} to stay near {@link Blocks#FARMLAND Farmland} when at it's job site.
+	 *
+	 * @param blocks The blocks.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder secondaryJobSites(Block... blocks);
+
+	/**
+	 * Blocks which might suffice as a job site for a Villager.
+	 *
+	 * <p>In vanilla this is used by the {@link VillagerProfession#FARMER Farmer} to stay near {@link Blocks#FARMLAND Farmland} when at it's job site.
+	 *
+	 * @param blocks A set of blocks.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder secondaryJobSites(Iterable<Block> blocks);
+
+	/**
+	 * Represents the sound made when a Villager works.
+	 * @param workSoundEvent The {@link SoundEvent} to be played.
+	 * @return this builder.
+	 */
+	VillagerProfessionBuilder workSound(SoundEvent workSoundEvent);
```

Comment:
I don't think represents is the best term here, maybe it could be "provides," or something similar?